utils/evm_loader.py

```diff
@@ -685,6 +687,46 @@ def deposit_wrapped_sol_from_solana_to_neon(self, solana_account, neon_account,
 
         self.send_token_from_solana_to_neon(solana_account, mint_pubkey, neon_account, full_amount, self.sol_chain_id)
 
+    def drain_wsol(
+        self,
+        withdraw_from: LocalAccount,
+        withdraw_to: Keypair,
+        withdraw_contract: Contract,
+        web3_client,
+        return_to: Keypair,
+    ):
+        neon_balance = web3_client.get_balance(withdraw_from)
+
+        ata = self.create_associate_token_acc(withdraw_to, withdraw_to, WRAPPED_SOL_MINT)
+        spl_token = SplToken(self, WRAPPED_SOL_MINT, TOKEN_PROGRAM_ID, withdraw_to)
+        ata_balance_before = int(spl_token.get_balance(ata, commitment=Confirmed).value.amount)
+        ata_info = spl_token.get_account_info(ata, commitment=Confirmed)
+        assert ata_info.owner == withdraw_to.pubkey()
+
+        tx = web3_client.make_raw_tx(from_=withdraw_from, amount=neon_balance)
+        value = (neon_balance - web3_client.eth.estimate_gas(tx) * web3_client.gas_price()) // 10**9
+        tx["value"] = value * 10**9
```

Comment:
did't get the logic 
first you do value // 10**9
and next line is  value * 10**9

Reply:
withdraw contract requires value to be divisible to 10**9
I drop remaining amount with // operation
Idk maybe it could be done better